File: Java/Która godzina.java
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        String bin1 = input.next();
        String bin2 = input.next();
        String bin3 = input.next();
        String bin4 = input.next();
        System.out.println(String.format("%02d:%02d", Integer.parseInt(bin1, 2) * 10 + Integer.parseInt(bin2, 2),Integer.parseInt(bin3, 2) * 10 + Integer.parseInt(bin4, 2)));
        input.close();
    }
}

File: C++/Która godzina.cpp
#include <string>
#include <cstdio>
#include <iostream>

int main() 
{
    char wynik[6];
    std::string bin1, bin2, bin3, bin4;
    std::cin >> bin1 >> bin2 >> bin3 >> bin4;
    std::printf("%02d:%02d\n", std::stoi(bin1, nullptr, 2) * 10 + std::stoi(bin2, nullptr, 2), std::stoi(bin3, nullptr, 2) * 10 + std::stoi(bin4, nullptr, 2));
    return 0;
}